code/modules/flufftext/Hallucination.dm

```diff
@@ -148,6 +148,8 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 
 /obj/effect/hallucination/simple/Moved(atom/OldLoc, Dir)
 	. = ..()
+	if(!loc)
+		return
```

Comment:
Show() is a bit weird but i dont think you need to early return for it, its nullsafe i think

Reply:
My concern is less null safety, and more the creation of references